src/components/ListItem.jsx

```diff
@@ -1,5 +1,25 @@
+import { updateItem } from '../api/firebase';
+import { isWithinLastDay } from '../utils/dates';
 import './ListItem.css';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item, listId }) {
+	const { id, name } = item;
+
+	const handlePurchase = async (e) => {
+		if (e.target.checked) {
+			await updateItem(listId, id);
+		}
+	};
+
+	return (
+		<li className="ListItem">
+			<input
+				type="checkbox"
+				id="mark-purchased"
+				onChange={handlePurchase}
+				checked={isWithinLastDay(item.dateLastPurchased)}
+			/>
+			<label htmlFor="mark-purchased">{name}</label>
```

Comment:
We need to take another look at this. The label appears semantic, but when I click on the item name, it doesn't automatically check the associated checkbox.

Reply:
Another point - now that we are able to add in "empty lists". I can see the checkbox on those null items too, which we need to hide.

![image](https://user-images.githubusercontent.com/99099243/235303146-a43757e1-e530-4b38-aa63-da4a36552fce.png)

Reply:
We’ll take a look at the label — thanks, Yao! Also with regard to the empty list items, if you are going to be filtering out the placeholder items in the API request, these would already be filtered out of the data no? Would this be fixed when we merge?

Reply:
Fixed the label with the latest commit. We can filter out the null items but I'm pretty sure that will happen when we merge.

Reply:
Great work guys! Yup, the id for each list item component ended up not being unique causing the issue. And you guys are absolutely right! Once we merge, we won't be displaying the null item so that will no longer be an issue.